Crash when upgrading from 2.0 to 3.2

### Terraform Version
v0.10.8
Also tested v0.10.6, same issue

### Affected Resource(s)

*Might* be related to storage account?

### Terraform Configuration Files
The code is quite long and split across several files and modules. I can upload extracts on demand if required. I'll just paste the container registry code as the crash looks related to it:

```hcl
resource "azurerm_storage_account" "storage_account" {
  name                      = "${var.storage_account_name}"
  resource_group_name       = "${var.resource_group_name}"
  location                  = "${var.location}"
  account_tier              = "Standard"
  account_replication_type  = "GRS"
}

resource "azurerm_container_registry" "container_registry" {
  name                = "${var.project}${var.env}cr"
  resource_group_name = "${var.resource_group_name}"
  location            = "${var.location}"
  admin_enabled       = true
  sku                 = "Basic"
  storage_account_id  = "${azurerm_storage_account.storage_account.id}"
}
```

### Debug Output
I've run with `TF_LOG=debug` but it shows several passwords and access keys.  I can upload extracts on demand if required.

### Panic Output

```
data.template_file.elk: Refreshing state...
azurerm_resource_group.resource_group: Refreshing state... (ID: /subscriptions/c27cfedb-f5e9-45e6-9642-...c94e7/resourceGroups/nomisapi-colin-rg)
azurerm_key_vault.vault: Refreshing state... (ID: /subscriptions/c27cfedb-f5e9-45e6-9642-...crosoft.KeyVault/vaults/nomisapi-colin)
azurerm_key_vault_secret.appgw_ssl_certificate: Refreshing state... (ID: https://nomisapi-colin.vault.azure.net/...icate/b058c83127b841c9a7247e166a916fe6)
azurerm_key_vault_secret.appgw_ssl_certificate_password: Refreshing state... (ID: https://nomisapi-colin.vault.azure.net/...sword/74c7dafe360e486dbd934077b9f3893d)
azurerm_key_vault_secret.bastion_ssh_private_key: Refreshing state... (ID: https://nomisapi-colin.vault.azure.net/...e-key/0d0e402b4a8440b0916c8bc62a4948ce)
azurerm_key_vault_secret.internal_ssh_private_key: Refreshing state... (ID: https://nomisapi-colin.vault.azure.net/...e-key/7a85023ecaee42b6950ae8d89428427d)
azurerm_key_vault_secret.bastion_ssh_public_key: Refreshing state... (ID: https://nomisapi-colin.vault.azure.net/...c-key/068cbbb7b00441879990515dba091857)
azurerm_key_vault_secret.internal_ssh_public_key: Refreshing state... (ID: https://nomisapi-colin.vault.azure.net/...c-key/c6b2728c34704318aa60c8ecfc3c4dbd)
azurerm_availability_set.availability_set: Refreshing state... (ID: /subscriptions/c27cfedb-f5e9-45e6-9642-...ilabilitySets/nomisapi-colin-appsrv-as)
azurerm_virtual_network.root-vnet: Refreshing state... (ID: /subscriptions/c27cfedb-f5e9-45e6-9642-...rtualNetworks/nomisapi-colin-root-vnet)
azurerm_public_ip.appgw_public_ip: Refreshing state... (ID: /subscriptions/c27cfedb-f5e9-45e6-9642-...dresses/nomisapi-colin-appgw-public-ip)
azurerm_availability_set.availability_set: Refreshing state... (ID: /subscriptions/c27cfedb-f5e9-45e6-9642-...ailabilitySets/nomisapi-colin-apigw-as)
azurerm_storage_account.storage_account: Refreshing state... (ID: /subscriptions/c27cfedb-f5e9-45e6-9642-...torage/storageAccounts/nomisapicolinsa)
azurerm_public_ip.public_ip: Refreshing state... (ID: /subscriptions/c27cfedb-f5e9-45e6-9642-...esses/nomisapi-colin-bastion-public_ip)
azurerm_container_registry.container_registry: Refreshing state... (ID: /subscriptions/c27cfedb-f5e9-45e6-9642-...nerRegistry/registries/nomisapicolincr)

Error: Error refreshing state: 7 error(s) occurred:

* module.networking.module.appgw-subnet-with-network-security-group.provider.azurerm: unexpected EOF
* module.bastion.module.vm.azurerm_public_ip.public_ip: 1 error(s) occurred:

* module.bastion.module.vm.azurerm_public_ip.public_ip: azurerm_public_ip.public_ip: unexpected EOF
* module.container-registry.azurerm_container_registry.container_registry: 1 error(s) occurred:

* module.container-registry.azurerm_container_registry.container_registry: azurerm_container_registry.container_registry: unexpected EOF
* module.networking.module.elk-subnet-with-network-security-group.provider.azurerm: unexpected EOF
* module.networking.module.apigw-subnet-with-network-security-group.provider.azurerm: unexpected EOF
* module.networking.module.appsrv-subnet-with-network-security-group.provider.azurerm: unexpected EOF
* module.networking.module.bastion-subnet-with-network-security-group.provider.azurerm: unexpected EOF


panic: interface conversion: interface {} is nil, not map[string]interface {}
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: goroutine 659 [running]:
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.updateV1ToV2StorageAccountName(0xc4203bca00, 0x21220c0, 0xc42115c000, 0x3, 0xc421ece548)
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_container_registry_migrate.go:78 +0x422
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.migrateAzureRMContainerRegistryStateV1toV2(0xc4203bca00, 0x21220c0, 0xc42115c000, 0x0, 0x0, 0xc4201c1e60)
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_container_registry_migrate.go:53 +0x117
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceAzureRMContainerRegistryMigrateState(0x1, 0xc4203bca00, 0x21220c0, 0xc42115c000, 0x2bea4c0, 0xc42092aa00, 0xc42092aa50)
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_container_registry_migrate.go:21 +0xb1
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc4212cf860, 0xc4203bca00, 0x21220c0, 0xc42115c000, 0xc421938980, 0x1, 0x8000000000d)
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:309 +0x2c6
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc4219f2540, 0xc4203bc9b0, 0xc4203bca00, 0x30d76c8, 0x0, 0xc42092ab60)
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:284 +0x9a
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc4213373c0, 0xc420eb0660, 0xc420eb0750, 0x0, 0x0)
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: reflect.Value.call(0xc4203a2060, 0xc42018d990, 0x13, 0x21ebca6, 0x4, 0xc42092af20, 0x3, 0x3, 0x0, 0x0, ...)
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/reflect/value.go:434 +0x906
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: reflect.Value.Call(0xc4203a2060, 0xc42018d990, 0x13, 0xc4204bc720, 0x3, 0x3, 0xc420fbc0c0, 0xc4204bc728, 0xc42274a0c0)
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: net/rpc.(*service).call(0xc42183a4c0, 0xc4211e05a0, 0xc421396140, 0xc420bbcd80, 0xc42164ca00, 0x1f9f180, 0xc420eb0660, 0x16, 0x1f9f1c0, 0xc420eb0750, ...)
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: created by net/rpc.(*Server).ServeCodec
2017-11-03T16:06:31.696Z [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
2017/11/03 16:06:31 [ERROR] root.bastion.vm: eval: *terraform.EvalRefresh, err: azurerm_public_ip.public_ip: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.elk-subnet-with-network-security-group: eval: *terraform.EvalConfigProvider, err: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.bastion.vm: eval: *terraform.EvalSequence, err: azurerm_public_ip.public_ip: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.elk-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.appsrv-subnet-with-network-security-group: eval: *terraform.EvalConfigProvider, err: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.container-registry: eval: *terraform.EvalRefresh, err: azurerm_container_registry.container_registry: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.appsrv-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.appgw-subnet-with-network-security-group: eval: *terraform.EvalConfigProvider, err: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.container-registry: eval: *terraform.EvalSequence, err: azurerm_container_registry.container_registry: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.appgw-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.appsrv-subnet-with-network-security-group: eval: *terraform.EvalOpFilter, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] [walkRefresh] Exiting eval tree: module.container-registry.azurerm_container_registry.container_registry
2017/11/03 16:06:31 [ERROR] root.networking.appsrv-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.appgw-subnet-with-network-security-group: eval: *terraform.EvalOpFilter, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] [walkRefresh] Exiting eval tree: module.networking.module.appsrv-subnet-with-network-security-group.provider.azurerm
2017/11/03 16:06:31 [ERROR] root.networking.elk-subnet-with-network-security-group: eval: *terraform.EvalOpFilter, err: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.elk-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] [walkRefresh] Exiting eval tree: module.networking.module.elk-subnet-with-network-security-group.provider.azurerm
2017/11/03 16:06:31 [ERROR] root.networking.appgw-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.azurerm_network_security_group.nsg"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.container-registry.output.registry_host"
2017/11/03 16:06:31 [ERROR] root.networking.bastion-subnet-with-network-security-group: eval: *terraform.EvalConfigProvider, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] [walkRefresh] Exiting eval tree: module.networking.module.appgw-subnet-with-network-security-group.provider.azurerm
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "output.REGISTRY_HOST"
2017/11/03 16:06:31 [TRACE] [walkRefresh] Exiting eval tree: module.bastion.module.vm.azurerm_public_ip.public_ip
2017/11/03 16:06:31 [ERROR] root.networking.bastion-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.output.nsg_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.azurerm_route_table.route-table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.azurerm_route_table.route-table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.azurerm_route_table.route-table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.bastion.module.vm.output.fqdn"
2017/11/03 16:06:31 [ERROR] root.networking.bastion-subnet-with-network-security-group: eval: *terraform.EvalOpFilter, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.output.route_table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.azurerm_network_security_group.nsg"
2017/11/03 16:06:31 [ERROR] root.networking.apigw-subnet-with-network-security-group: eval: *terraform.EvalConfigProvider, err: unexpected EOF
2017/11/03 16:06:31 [ERROR] root.networking.apigw-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.azurerm_route.blackhole-102"
2017/11/03 16:06:31 [ERROR] root.networking.apigw-subnet-with-network-security-group: eval: *terraform.EvalOpFilter, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.appsrv_subnet_nsg_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.output.route_table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-ssh-from-bastion-to-appsrv"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.azurerm_route.blackhole-101"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.azurerm_route.blackhole-102"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.azurerm_route.blackhole-101"
2017/11/03 16:06:31 [ERROR] root.networking.bastion-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.var.nsg_name"
2017/11/03 16:06:31 [TRACE] [walkRefresh] Exiting eval tree: module.networking.module.bastion-subnet-with-network-security-group.provider.azurerm
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.azurerm_subnet.subnet"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.output.nsg_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.azurerm_route.blackhole-101"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.azurerm_route.blackhole-102"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.azurerm_network_security_rule.inbound-allow-subnet-to-app-port"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.output.address_prefix"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.output.nsg_id"
2017/11/03 16:06:31 [ERROR] root.networking.apigw-subnet-with-network-security-group: eval: *terraform.EvalSequence, err: unexpected EOF
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.appsrv_subnet_nsg_id"
2017/11/03 16:06:31 [TRACE] [walkRefresh] Exiting eval tree: module.networking.module.apigw-subnet-with-network-security-group.provider.azurerm
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.output.nsg_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.azurerm_subnet.subnet"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.azurerm_route_table.route-table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-internet-to-appgw-port"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.output.address_prefix"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.azurerm_route.blackhole-102"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.azurerm_network_security_rule.inbound-deny-all"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.output.subnet_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-internet-to-backend-health-api-port"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.output.route_table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.azurerm_route.blackhole-101"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.output.id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.azurerm_network_security_group.nsg"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.output.route_table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.appgw_subnet_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_route.transit-nomis"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "output.APPGW_SUBNET_NAME"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.azurerm_network_security_rule.inbound-allow-lb-to-app-port"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.azurerm_network_security_rule.inbound-deny-all"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.output.nsg_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.output.id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.azurerm_subnet.subnet"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.output.nsg_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.apigw_subnet_nsg_id"
2017-11-03T16:06:31.702Z [DEBUG] plugin: plugin process exited: path=/Users/colinsaliceti/Documents/git/nomis-api-terraform-azure/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v0.3.2_x4
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.azurerm_network_security_group.nsg"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-ssh-from-bastion-to-apigw"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-from-app-gw-subnet-to-app-port"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.appsrv_subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.output.nsg_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.apigw_subnet_nsg_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.output.id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-vm.var.subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.var.subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.output.nsg_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.azurerm_network_security_rule.inbound-deny-all"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-ssh-from-whitelist-to-bastion"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appsrv-subnet-with-network-security-group.output.subnet_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.azurerm_lb.load-balancer"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.output.load_balancer_private_ip"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.azurerm_lb_probe.probe"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.apigw_subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.azurerm_lb_backend_address_pool.be-address-pool"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.azurerm_dns_a_record.appsrv"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.output.load_balancer_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.output.be_address_pool_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.appgw-subnet-with-network-security-group.azurerm_network_security_rule.inbound-deny-all"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-vm.var.lb_backend_pool_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-appgw-to-subnet"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.azurerm_route_table.route-table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-vm.azurerm_network_interface.interface"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.azurerm_subnet.subnet"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.azurerm_network_security_group.nsg"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-vm.output.private_ips"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.output.subnet_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.azurerm_route.blackhole-102"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.output.route_table"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.output.address_prefix"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.apigw-subnet-with-network-security-group.output.subnet_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.output.id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.output.nsg_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-load-balancer.azurerm_lb_rule.lb-rule"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.apigw-vm.var.subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.bastion_subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-http-from-bastion"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-logstash-from-appsrv"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-ssh-from-bastion-to-elk"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.apigw-vm.azurerm_network_interface.interface"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.azurerm_network_security_rule.inbound-deny-all"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.bastion.var.subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.azurerm_network_security_rule.inbound-allow-logstash-from-apigw"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.appsrv-vm.azurerm_virtual_machine.vm"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.bastion.module.vm.var.subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.output.nsg_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.apigw-vm.azurerm_virtual_machine.vm"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.azurerm_subnet.subnet"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.azurerm_route.blackhole-101"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.bastion-subnet-with-network-security-group.output.address_prefix"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.elk_subnet_nsg_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.apigw-vm.output.private_ips"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "output.APIGW_PRIVATE_IPS"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.output.subnet_name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.bastion.module.vm.azurerm_network_interface.interface"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.output.address_prefix"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.module.elk-subnet-with-network-security-group.output.id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.networking.output.elk_subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.bastion.module.vm.azurerm_virtual_machine.vm"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.elk.var.subnet_id"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.bastion.module.vm.output.name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.elk.azurerm_network_interface.interface"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.bastion.output.vm-name"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.bastion.azurerm_dns_cname_record.bastion"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "output.BASTION"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "module.elk.azurerm_virtual_machine.vm"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "provider.azurerm (close)"
2017/11/03 16:06:31 [TRACE] dag/walk: upstream errored, not walking "root"
2017/11/03 16:06:31 [DEBUG] plugin: waiting for all plugin processes to complete...
2017-11-03T16:06:31.703Z [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2017-11-03T16:06:31.705Z [DEBUG] plugin: plugin process exited: path=/Users/colinsaliceti/Documents/git/nomis-api-terraform-azure/.terraform/plugins/darwin_amd64/terraform-provider-template_v1.0.0_x4
```

### Expected Behavior
`No changes. Infrastructure is up-to-date.`

### Actual Behavior
`!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!`

### Steps to Reproduce

1. `terraform plan` is successful with:

```
provider "azurerm" {
  version = "=0.2.0"
...
}
```
2. Pin azurerm to 0.3.2:

```
provider "azurerm" {
  version = "=0.3.2"
...
}
```
3. Make required syntax changes to solve:

```
Error: azurerm_key_vault.vault: expected access_policy.9.secret_permissions.0 to be one of [backup delete get list purge recover restore set], got all
Error: module.container-registry.azurerm_storage_account.storage_account: "account_replication_type": required field is not set
Error: module.container-registry.azurerm_storage_account.storage_account: "account_tier": required field is not set
```
4. `terraform plan` crashes

Hi, I'm getting the exact same issue when upgrading 0.1.7 to 0.3.1 or 0.3.2. Works ok on 0.2.x. 

```
panic: runtime error: invalid memory address or nil pointer dereference
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1ed879f]
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4:
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: goroutine 890 [running]:
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.flattenAzureRmVirtualMachineScaleSetNetworkProfile(0xc4205783e0, 0x220101a, 0x17, 0x1fbe440)
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_virtual_machine_scale_set.go:903 +0x27f
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmVirtualMachineScaleSetRead(0xc4202fc0e0, 0x21220c0, 0xc420546000, 0x0, 0x2bea4c0)
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_virtual_machine_scale_set.go:733 +0xcc3
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc42050fc80, 0xc420166280, 0x21220c0, 0xc420546000, 0xc42037f678, 0xc420054701, 0x80001051ea0)
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:321 +0x199
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc42014ad90, 0xc420166230, 0xc420166280, 0x30d7000, 0x0, 0xc4203c0a80)
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:284 +0x9a
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc4204c0be0, 0xc4205798b0, 0xc4205799e0, 0x0, 0x0)
2017-11-07T14:16:00.091-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017-11-07T14:16:00.092-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: reflect.Value.call(0xc42037c060, 0xc42000d6c0, 0x13, 0x21ebca6, 0x4, 0xc4202d6f20, 0x3, 0x3, 0x0, 0x0, ...)
2017-11-07T14:16:00.092-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/reflect/value.go:434 +0x906
2017-11-07T14:16:00.092-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: reflect.Value.Call(0xc42037c060, 0xc42000d6c0, 0x13, 0xc42002d720, 0x3, 0x3, 0xc42034c540, 0xc4201c06c0, 0xc4200800a0)
2017-11-07T14:16:00.092-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/reflect/value.go:302 +0xa4
2017-11-07T14:16:00.092-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: net/rpc.(*service).call(0xc4204bd0c0, 0xc42056ca00, 0xc4203957f0, 0xc42015a580, 0xc4204c1160, 0x1f9f180, 0xc4205798b0, 0x16, 0x1f9f1c0, 0xc4205799e0, ...)
2017-11-07T14:16:00.092-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/net/rpc/server.go:381 +0x142
2017-11-07T14:16:00.092-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: created by net/rpc.(*Server).ServeCodec
2017-11-07T14:16:00.092-0500 [DEBUG] plugin.terraform-provider-azurerm_v0.3.2_x4: 	/usr/local/go/src/net/rpc/server.go:475 +0x36b
```

--- a/azurerm/resource_arm_container_registry_migrate.go
+++ b/azurerm/resource_arm_container_registry_migrate.go
@@ -75,8 +75,13 @@ func updateV1ToV2StorageAccountName(is *terraform.InstanceState, meta interface{
 	}
 
 	inputAccounts := result.Value.([]interface{})
-	inputAccount := inputAccounts[0].(map[string]interface{})
-	name := inputAccount["name"].(string)
+	inputAccount := inputAccounts[0]
+	if inputAccount == nil {
+		return nil
+	}
+
+	account := inputAccount.(map[string]interface{})
+	name := account["name"].(string)
 	storageAccountId, err := findAzureStorageAccountIdFromName(name, meta)
 	if err != nil {
 		return err

